src/Chainweb/Cut.hs

```diff
@@ -408,19 +555,32 @@ data Join a = Join
     , _joinQueue :: !(JoinQueue a)
     }
 
+-- | This computes the join for cuts accross all chains.
```

Comment:
Can we get some docs on the `Join` datatype?

Reply:
Let's postpone this. It's a rabbit whole. But feel free to remind me to not forget about it. Better documenting the theory behind cuts and the implementation of it has been on my TODO for a while now.

Actually, `Chainweb.Test.Cut` has quite a bit of documentation around the properties of cuts and in particular the lattice structure. But it's very technical and hard to understand without more background at this point.

Reply:
I added brief comments to the data type and its constructor fields. But I am not sure how helpful that actually is.